pkg/rpc/rpc.go

```diff
@@ -86,14 +85,19 @@ func (r *RPC) Call(method string, params json.RawMessage) (json.RawMessage, erro
 	r.requestsMu.Unlock()
 
 	r.encoderMu.Lock()
```

Comment:
Might be nit-picky, but there are three following unlock cases which makes this a bit hard to read. Even as I dislike kind of "useless" or "inline-able" methods, maybe consider refactoring this into a helper method where you can just `defer` the unlocking and return the error?